main.php

```diff
@@ -27,12 +27,11 @@ function ($errno, $errstr, $errfile, $errline, array $errcontext) {
                     $fs->mkdir($dataFolder . "/out/files" . $subfolder);
                 }
             }
+            $destinationPath = $dataFolder . "/out/files" . $subfolder . "/" . $sourceFile->getBasename();
```

Comment:
Na řádku 24 je `$subfolder` jako "cesta až po koncové lomítko za `in/files/`. Ale tady se spojuje bez lomítka.

Reply:
zrovna na to koukam a premyslim jaktoze to funguje :)

Reply:
Jo, ono to vlastně vyjde, protože délka stringu vs. pole indexované od nuly! Ale přesně kvůli tomuhle jsem chtěl udělal tu "filepath knihovnu", která tohle všechno odabstrahuje... přijde mi to potenciálně nebezpečné.

Reply:
co by ta knihovna měla dělat?

Reply:
A ono samozřejmě něco už existuje! https://github.com/eloquent/pathogen :)

Reply:
ha, mrknu na to, nevymění to akorát jeden nesrozumitelnej kód za jinej (plnej `array_slice`)?

Reply:
```
use Eloquent\Pathogen\Path;

$path = Path::fromString('/path/to/foo/bar');
$pathWithReplacement = $path->replace(1, array('for', 'baz'), 2);

echo $pathWithReplacement->string(); // outputs '/path/for/baz/bar'
```

ale moc srouzmitelny mi to neprijde :)

Reply:
Tak možnosti jsou i jiné https://github.com/webmozart/path-util

Reply:
Z toho by šlo použít akorát tak
```
Path::join('phar://C:/Documents', 'projects/my-project.phar', 'composer.json');
// => phar://C:/Documents/projects/my-project.phar/composer.json
```
o to jde?

Reply:
Já osobně bych byl pro víc komplexní přístup (tj. ten Pathogen - i když uznávám, že ideální to není), ale tohle spojování složek do cesty je věc, která je v tomhle případě nesrozumitelná a problematická. A dělá se v každé komponentě prakticky...

Reply:
no jestli napíšu 
```
$path = $var1 . "/" . $var2 . "/" . $var3;
```
nebo
```
$path = Path::join($var1, $var2, $var3);
```
je imho jedno, nesrozumitelný to bude stejně. Můžu zkusit ty proměnný líp pojmenovat a nedávat tam žádný stringy, jen proměnný?

nejkomplikovanější konstrukt tam je

```
$subfolder = substr($sourceFile->getPath(), strlen($dataFolder . 'in/files/'));
```
a s tím ani jedna z těch knihoven nepomůže, jenom se to místo operace se stringama přesune do operací s poli.

Reply:
jo, to spojovani mi nevadi, ale ten substr je podle me to stejny co dela `$sourceFile->getRelativePath()`

Reply:
není, ten substr odřízne root datafolderu a `in/files/`, tj. `/tmp/mytmpfolder/data/in/files/myfolder/mysubfolder` => `myfolder/mysubfolder`

Reply:
no vsak

Reply:
relativne k in($dataFolder . "/in/files");

Reply:
jojo, ale u `SplFileInfo::getRelativePath()` nejde zadat k čemu relativní to má bejt, ne?

Reply:
nejde, je relativni k tomu `in()`, coz je presne to co chces imho

Reply:
Tak ten pathogen má neúplnou a místy zavádějící dokumentaci.

Reply:
@odinuv co je to `in()`?

Reply:
$finder->name("*.gz")->in($dataFolder . "/in/files");

Reply:
Aha!

Reply:
zkusil jsem to ještě trochu přepsat, zatím bez pathogenu, mrkněte, jestli to je srozumitelnější.

Reply:
Taky jsem neznal, díky 👍